src/realm/array.hpp

```diff
@@ -703,21 +545,7 @@ class Array : public Node, public ArrayParent {
     friend class Allocator;
     friend class SlabAlloc;
     friend class GroupWriter;
-
-    // Optimized implementation for release mode
-    template <class cond, size_t bitwidth, class Callback>
-    bool find_optimized(int64_t value, size_t start, size_t end, size_t baseindex, QueryStateBase* state,
-                        Callback callback) const;
-
-protected:
-    // Called for each search result
-    template <class Callback>
-    bool find_action(size_t index, util::Optional<int64_t> value, QueryStateBase* state, Callback callback) const;
-
-    bool find_action_pattern(size_t index, uint64_t pattern, QueryStateBase* state) const;
-    template <size_t bitwidth, class Callback>
-    bool find_all_will_match(size_t start, size_t end, size_t baseindex, QueryStateBase* state,
-                             Callback callback) const;
+    friend class ArrayWithFind;
```

Comment:
Why is this necessary if ArrayWithFind inherits from Array? Are there private members that ArrayWithFind should have access to but not other children of Array?

Reply:
Well - it must have been needed at some point :-) But apparently not any more.

Reply:
Ok, looks like it is still needed in one place according to CI. If you can find an easy way to remove the friendship requirement from this one place that would be ideal, but if not no worries it is not a big issue and not blocking for this PR.

```
/mnt/jenkins/workspace/realm_realm-core_PR-5015/src/realm/array_with_find.hpp:871:26: error: 'm_width' is a protected member of 'realm::Array'
    size_t fw = foreign->m_width;
```